lib/ansible/modules/cloud/amazon/ec2_lc.py

```diff
@@ -188,6 +194,13 @@ def create_launch_config(connection, module):
     classic_link_vpc_security_groups = module.params.get('classic_link_vpc_security_groups')
     bdm = BlockDeviceMapping()
 
+    if user_data_path:
+        try:
+            user_data_file = open(user_data_path, 'r')
```

Comment:
When reading files, please use a `with` block so the file will always be closed. This code will never close the filehandle.